cli/internal/cmd/upgradeapply.go

```diff
@@ -180,19 +316,37 @@ func parseUpgradeApplyFlags(cmd *cobra.Command) (upgradeApplyFlags, error) {
 		return upgradeApplyFlags{}, fmt.Errorf("parsing force argument: %w", err)
 	}
 
-	return upgradeApplyFlags{configPath: configPath, yes: yes, upgradeTimeout: timeout, force: force}, nil
+	logLevelString, err := cmd.Flags().GetString("tf-log")
+	if err != nil {
+		return upgradeApplyFlags{}, fmt.Errorf("parsing tf-log string: %w", err)
+	}
+	logLevel, err := terraform.ParseLogLevel(logLevelString)
+	if err != nil {
+		return upgradeApplyFlags{}, fmt.Errorf("parsing Terraform log level %s: %w", logLevelString, err)
+	}
+
+	return upgradeApplyFlags{
+		configPath:        configPath,
+		yes:               yes,
+		upgradeTimeout:    timeout,
+		force:             force,
+		terraformLogLevel: logLevel,
+	}, nil
 }
 
 type upgradeApplyFlags struct {
-	configPath     string
-	yes            bool
-	upgradeTimeout time.Duration
-	force          bool
+	configPath        string
+	yes               bool
+	upgradeTimeout    time.Duration
+	force             bool
+	terraformLogLevel terraform.LogLevel
 }
 
 type cloudUpgrader interface {
 	UpgradeNodeVersion(ctx context.Context, conf *config.Config) error
 	UpgradeHelmServices(ctx context.Context, config *config.Config, timeout time.Duration, allowDestructive bool) error
 	UpdateMeasurements(ctx context.Context, newMeasurements measurements.M) error
 	GetClusterMeasurements(ctx context.Context) (measurements.M, *corev1.ConfigMap, error)
+	PlanTerraformMigrations(ctx context.Context, opts kubernetes.TerraformUpgradeOptions) (bool, error)
+	ApplyTerraformMigrations(ctx context.Context, file file.Handler, opts kubernetes.TerraformUpgradeOptions) error
```

Comment:
Is there something that speaks against putting these two into a separate type?
I think it would make sense to have the implementation of those two functions inside `cli/internal/terraform` and not in `cli/internal/kubernetes`. The functions do not interact with the cluster, but with the platform the cluster is running on.

Reply:
I think they should not be placed in `cli/internal/terraform` as their scope goes beyond the interaction with Terraform only. E.g., `ApplyTerraformMigrations` formats the Terraform output and writes it to the `clusterid.File`, while the `terraform` package only wraps singular Terraform operations. (See [cluster creation](https://github.com/edgelesssys/constellation/blob/fd83f3439e847ff14a51525f7bd9f55133588671/cli/internal/cloudcmd/create.go#L169) as a reference)

Reply:
Okay. Then lets put them into a new pkg: `cli/internal/upgrade`. Looks like the implementations don't depend on other parts of `Upgrader`. Sorry about initially pointing you towards adding the methods to that interface. But looking at it for a second time I think it makes sense to separate the terraform migrations from the kubernetes operations..